packages/theme-check-common/src/checks/valid-block-target/index.spec.ts

```diff
@@ -479,6 +500,351 @@ describe('Module: ValidBlockTarget', () => {
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+          'sections/slideshow.liquid': `
+            {% schema %}
+            {
+              "name": "Slideshow",
+              "blocks": [
+                {
+                  "type": "slide"
+                }
+              ],
+              "presets": [
+                {
+                  "name": "Default",
+                  "blocks": {
+                    "slide_hash": {
+                      "type": "slide",
+                      "blocks": {
+                        "group_hash": {
+                           "type": "group"
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+        };
+
+        const offenses = await check(theme, [ValidBlockTarget]);
+        expect(offenses).to.have.length(1);
+        expect(offenses[0].message).to.equal(
+          'Block type "group" is not allowed in "slide" blocks. Allowed types are: text, image.',
+        );
+      });
+
+      it(`should report errors for further nested blocks when they are not allowed (${path} bucket)`, async () => {
+        const theme: MockTheme = {
+          'blocks/image.liquid': '',
+          'blocks/group.liquid': `
+            {% schema %}
+            {
+              "name": "Group",
+              "blocks": [
+                {
+                  "type": "text"
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+          'blocks/text.liquid': '',
+          'blocks/slide.liquid': `
+            {% schema %}
+            {
+              "name": "Slide",
+              "blocks": [
+                {
+                  "type": "group"
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+          'sections/slideshow.liquid': `
+            {% schema %}
+            {
+              "name": "Slideshow",
+              "blocks": [
+                {
+                  "type": "slide"
+                }
+              ],
+              "presets": [
+                {
+                  "name": "Default",
+                  "blocks": [
+                    {
+                      "type": "slide",
+                      "blocks": [
+                        {
+                          "type": "group",
+                          "blocks": [
+                            {
+                              "type": "image"
+                            }
+                          ]
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+        };
+
+        const offenses = await check(theme, [ValidBlockTarget]);
+        expect(offenses).to.have.length(1);
+        expect(offenses[0].message).to.equal(
```

Comment:
Given the cross file added complexity, can you also verify the offence `uri` surfaces this error for the right theme file?

Reply:
Good idea!